src/provider/proxyprovider.cpp

```diff
@@ -0,0 +1,86 @@
+#include "provider/proxyprovider.h"
+#include "apiresponse.h"
+#include "httpwrapper.h"
+
+namespace internal
+{
+    bool isRequestSuccessful(ErdGenericApiResponse const &response)
+    {
+        std::string const error = response.getError();
+        std::string const code = response.getCode();
+
+        return error.empty() && (code.find("success")!= std::string::npos);
+    }
+
+    nlohmann::json getDataIfValid(wrapper::http::Result const &res)
+    {
+        if (res.error)
+        {
+            throw std::runtime_error(res.statusMessage);
+        }
+
+        ErdGenericApiResponse wrappedResponse(res.body);
+
+        if (res.status != STATUS_CODE_OK || !isRequestSuccessful(wrappedResponse))
+        {
+            throw std::runtime_error(ERROR_MSG_HTTP_REQUEST_FAILED + res.statusMessage + ". " +
+                                     ERROR_MSG_REASON + wrappedResponse.getError());
+        }
+
+        return wrappedResponse.getData<nlohmann::json>();
+    }
+
+}
+
+ProxyProvider::ProxyProvider(std::string url):
+    m_url(std::move(url)) {}
+
+Account ProxyProvider::getAccount(Address const &address)
+{
+    wrapper::http::Client client(m_url);
+    wrapper::http::Result const result = client.get("/address/" + address.getBech32Address());
+
+    auto data = internal::getDataIfValid(result);
+
+    if (!data.contains("account"))
```

Comment:
`requireAttribute()`, utility function? However, I think these assertions are a bit redundant. If the API responded with success, then it didn't lie :)

Reply:
After yesterday's discussion, we decided to check anyway for these fields. I've added `requireAttribute` utility function, however.